crates/ruff_python_parser/src/parser/statement.rs

```diff
@@ -1181,27 +1183,158 @@ impl<'src> Parser<'src> {
+        let start = self.node_start();
+        self.bump(TokenKind::Except);
+
+        let block_kind = if self.eat(TokenKind::Star) {
+            ExceptClauseKind::Star
+        } else {
+            ExceptClauseKind::Normal
+        };
+
+        let type_ = if self.at_expr() {
+            // test_err except_stmt_invalid_expression
+            // try:
+            //     pass
+            // except yield x:
+            //     pass
+            // try:
+            //     pass
+            // except* *x:
+            //     pass
+            let parsed_expr = self.parse_expression_list(AllowStarredExpression::No);
+            if matches!(
+                parsed_expr.expr,
+                Expr::Tuple(ast::ExprTuple {
+                    parenthesized: false,
+                    ..
+                })
+            ) {
+                // test_err except_stmt_unparenthesized_tuple
+                // try:
+                //     pass
+                // except x, y:
+                //     pass
+                // except x, y as exc:
+                //     pass
+                // try:
+                //     pass
+                // except* x, y:
+                //     pass
+                // except* x, y as eg:
+                //     pass
+                self.add_error(
+                    ParseErrorType::OtherError(
+                        "Multiple exception types must be parenthesized".to_string(),
+                    ),
+                    &parsed_expr,
+                );
+            }
+            Some(Box::new(parsed_expr.expr))
+        } else {
+            if block_kind.is_star() || self.at(TokenKind::As) {
+                // test_err except_stmt_missing_exception
+                // try:
+                //     pass
+                // except as exc:
+                //     pass
+                // # If a '*' is present then exception type is required
+                // try:
+                //     pass
+                // except*:
+                //     pass
+                // except*
+                //     pass
+                // except* as exc:
+                //     pass
+                self.add_error(
+                    ParseErrorType::OtherError("Expected one or more exception types".to_string()),
+                    self.current_token_range(),
+                );
+            }
+            None
+        };
+
+        let name = if self.eat(TokenKind::As) {
+            if self.at(TokenKind::Name) {
+                Some(self.parse_identifier())
+            } else {
+                // test_err except_stmt_missing_as_name
+                // try:
+                //     pass
+                // except Exception as:
+                //     pass
+                // except Exception as
+                //     pass
+                self.add_error(
+                    ParseErrorType::OtherError("Expected symbol name after `as`".to_string()),
```

Comment:
Nit: As a user it wouldn't be clear what symbol refers to. The documentation refers to the part after `as` as the target, but I think that might still be hard to understand. Maybe just use `Expected name after as` or `Expected variable name after `as`?